### PR TITLE
#27 convert maven build scripts to gradle

### PR DESCRIPTION
**Pull request by adamkorynta**
## Problem Description
Maven offers very little flexibility for building applications. HEC has standardized their libraries to now use Gradle instead.

## Solution

Convert maven build to gradle.

Replace the compile task with gradle's compile.
Replace the easy-pack maven plugin with gradle's tar task.
Add in a JavaExec task for running within an IDE.

## how you tested the change

Testing through IntelliJ IDE though tasks are IDE agnostic.

## Where the following done:

- [ ] Tests. Check all that apply:
   - [ ] Unit tests created or modified that run during ant test.
   - [ ] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [X] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+jersey-common = { module = "org.glassfish.jersey.core:jersey-common", version.ref = "jersey" }
+jersey-container-servlet = { module = "org.glassfish.jersey.containers:jersey-container-servlet", version.ref = "jersey" }
+jersey-hk2 = { module = "org.glassfish.jersey.inject:jersey-hk2", version.ref = "jersey" }
+postgresql = { module = "org.postgresql:postgresql", version.ref = "postgresql" }
+swagger-jaxrs2 = { module = "io.swagger.core.v3:swagger-jaxrs2", version.ref = "swagger" }
+swagger-ui = { module = "org.webjars:swagger-ui", version.ref = "swagger-ui" }
+jaxb-runtime = { module = "org.glassfish.jaxb:jaxb-runtime", version.ref = "glassfish-jaxb" }
+
+#Test Dependencies
+junit-api = { module = "org.junit.jupiter:junit-jupiter-api", version.ref = "junit" }
```
Comment (MikeNeilson) [inline gradle/libs.versions.toml:36]: The platform is missing.

File: src/main/java/org/opendcs/odcsapi/dao/ApiComputationDAO.java [hunk 2]
```diff
@@ -522,7 +522,7 @@ else if (name.toLowerCase().endsWith("_missing"))
 	}
 		
 	/**
-	 * Create computation object & fill with current row of result set.
+	 * Create computation object and fill with current row of result set.
```
Comment (MikeNeilson) [inline src/main/java/org/opendcs/odcsapi/dao/ApiComputationDAO.java:525]: alternatively `&amp;` (for javadoc)

File: src/main/java/org/opendcs/odcsapi/dao/ApiDaoBase.java [hunk 3]
```diff
@@ -137,7 +137,7 @@ public ResultSet doQuery2(String q)
 	* Executes an UPDATE or INSERT query.
 	* Thread safe: internally synchronized on the modify-statement.
 	* @param q the query string
-	* @throws DatabaseException  if the update fails.
+	* @throws DbException  if the update fails.
```
Comment (MikeNeilson) [inline src/main/java/org/opendcs/odcsapi/dao/ApiDaoBase.java:140]: I might be worth actually making this DatabaseException to be in line with opendcs itself.

granted likely a thing for a different PR.

File: src/main/java/org/opendcs/odcsapi/dao/ApiComputationDAO.java (hunk 2 above)
Comment (adamkorynta) [reply src/main/java/org/opendcs/odcsapi/dao/ApiComputationDAO.java:525]: yeah, I just didn't find a case where that would be nicer to see than "and"

**Comment by adamkorynta** [comment]
I rebased this PR on: https://github.com/opendcs/rest_api/pull/47 for simplicity.

**Comment by adamkorynta** [comment]
The `./war-files/META-INF/context.xml` looks like an example rather than something we would want to bundle in any distribution. I'm not sure the best place to put that at the moment. Under an `./examples` directory?

File: settings.gradle [hunk 4]
```diff
@@ -0,0 +1,10 @@
+rootProject.name = "opendcs_rest_api"
+
+include "rest_api"
+include "web_client"
+include "odcsapi-jetty-standalone"
+include "odcs-web-client-jetty-standalone"
+project(':rest_api').name = 'odcsapi'
```
Comment (adamkorynta) [inline settings.gradle:7]: I'll rename these directories in a follow-up PR

**Comment by adamkorynta** [comment]
I rebased onto: https://github.com/opendcs/rest_api/pull/48

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**